html/dom/documents/dom-tree-accessors/named-item.html

```diff
@@ -0,0 +1,643 @@
+  "br",
+  "button",
+  "canvas",
+  "caption",
+  "cite",
+  "code",
+  "col",
+  "colgroup",
+  "data",
+  "datalist",
+  "dd",
+  "del",
+  "details",
+  "dfn",
+  "dialog",
+  "div",
+  "dl",
+  "dt",
+  "em",
+  "fieldset",
+  "figcaption",
+  "figure",
+  "footer",
+  "h1",
+  "h2",
+  "h3",
+  "h4",
+  "h5",
+  "h6",
+  "head",
+  "header",
+  "hgroup",
+  "hr",
+  "html",
+  "i",
+  "input",
+  "ins",
+  "kbd",
+  "label",
+  "legend",
+  "li",
+  "link",
+  "main",
+  "map",
+  "mark",
+  "math",
+  "menu",
+  "meta",
+  "meter",
+  "nav",
+  "noscript",
+  "ol",
+  "optgroup",
+  "option",
+  "output",
+  "p",
+  "param",
+  "picture",
+  "pre",
+  "progress",
+  "q",
+  "rp",
+  "rt",
+  "ruby",
+  "s",
+  "samp",
+  "script",
+  "section",
+  "select",
+  "slot",
+  "small",
+  "source",
+  "span",
+  "strong",
+  "style",
+  "sub",
+  "summary",
+  "sup",
+  "svg",
+  "table",
+  "tbody",
+  "td",
+  "template",
+  "textarea",
+  "tfoot",
+  "th",
+  "thead",
+  "time",
+  "title",
+  "tr",
+  "track",
+  "u",
+  "ul",
+  "var",
+  "video",
+  "wbr",
+  "unsupportedelement",
+  "acronym",
+  "bgsound",
+  "dir",
+  "noframes",
+  "isindex",
+  "keygen",
+  "listing",
+  "menuitem",
+  "nextid",
+  "noembed",
+  "plaintext",
+  "rb",
+  "rtc",
+  "strike",
+  "xmp",
+  "basefont",
+  "big",
+  "blink",
+  "center",
+  "font",
+  "marquee",
+  "multicol",
+  "nobr",
+  "spacer",
+  "tt",
+];
+
+function runDefaultTest(win = window, root = testDiv) {
+  const doc = win.document;
+
+  let windowExpected = new Map();
+  let documentExpected = new Map();
+  let windowUnexpected = new Set();
+  let documentUnexpected = new Set();
+
+  const addToWindow = (key, val) => {
+    if (!windowExpected.has(key)) {
+      windowExpected.set(key, [val]);
+    } else if (Array.isArray(windowExpected.get(key))) {
+      windowExpected.get(key).push(val);
+    } // else it's present but not an array, so it's a WindowProxy, so we
+      // don't expect val to be on the window.
+  };
+
+  const addToDocument = (key, val) => {
+    if (!documentExpected.has(key)) {
+      documentExpected.set(key, [val]);
+    } else {
+      documentExpected.get(key).push(val);
+    }
+  };
+
+  for (const elem of root.querySelectorAll("*")) {
+    // We don't use the IDL attributes in case the browser doesn't support the
+    // element at all (like applet at the time of writing).
+    if (elem.hasAttribute("id")) {
+      const id = elem.getAttribute("id");
+      if (id === "") {
+        windowUnexpected.add(id);
+        documentUnexpected.add(id);
+      } else {
+        if (isArrayIndexPropertyName(id)) {
+          // WindowProxy has special behavior for these
+          // https://html.spec.whatwg.org/#windowproxy-getownproperty
+          windowUnexpected.add(id);
+        } else {
+          addToWindow(id, elem);
+        }
+        if (elem.localName == "applet" || elem.localName == "object" ||
+            (elem.localName == "img" && elem.name !== null &&
```

Comment:
Good catch.  I assumed it worked the same as getAttribute() and returned null for an absent attribute.